### PR TITLE
update cinch submodule

### PR DESCRIPTION
**Pull request by ryosukep**
change some directory structure and cmake-files to accommodate new cinch.

### CODE REVIEW

File: flecsi-sp/minimal/mesh.h [hunk 1]
```diff
@@ -122,15 +161,21 @@ class minimal_mesh_t
   size_t
   indices(
     size_t index_space_id
+<<<<<<< HEAD
```
Comment (cmsquared) [inline flecsi-sp/minimal/mesh.h]: I don't think the failures are because of this, but this _will_ cause a failure.

Comment (cmsquared) [inline flecsi-sp/minimal/mesh.h]: As @charest pointed out to me yesterday, the build is likely using a docker image that has an installed `flecsi` that was built from the `master` branch, which is currently broken.  We'll need to either fix this in the docker container or change which branch we pull from for `flecsi`

File: config/project.cmake [hunk 2]
```diff
@@ -38,11 +38,12 @@ else()
   message(STATUS "Note: using 32 bit integer ids.")
 endif()
 
+cinch_load_extras()
 #------------------------------------------------------------------------------#
 # Add library targets
 #------------------------------------------------------------------------------#
 
-cinch_add_library_target(flecsi-sp specializations)
+cinch_add_library_target(flecsi-sp flecsi-sp)
```
Comment (junghans) [inline config/project.cmake]: Why do you want to move these files around?

And if you do that you will need to replace specializations in `doc/doxygen.conf.in` as well.

File: config/project.cmake [hunk 3]
```diff
@@ -51,6 +52,17 @@ list( APPEND FleCSI_SP_LIBRARIES flecsi-sp )
 
 set(CINCH_HEADER_SUFFIXES "\\.h")
 
+#cinch_load_extras()
```
Comment (junghans) [inline config/project.cmake]: Remove.

File: config/project.cmake [hunk 4]
```diff
@@ -51,6 +52,17 @@ list( APPEND FleCSI_SP_LIBRARIES flecsi-sp )
...
+
+#------------------------------------------------------------------------------#
+# FleCSI Library
+#------------------------------------------------------------------------------#
+
+find_package(FleCSI REQUIRED)
+MESSAGE( STATUS ${FleCSI_LIBRARIES} )
+list( APPEND FleCSI_SP_LIBRARIES ${FleCSI_LIBRARIES} )
+include_directories(${FleCSI_INCLUDE_DIR})
+
```
Comment (junghans) [inline config/project.cmake]: What about the rest of `packages.cmake`? (e.g. cxx14 support)

File: flecsi-sp/minimal/mesh.h [hunk 5]
```diff
@@ -51,6 +58,10 @@ class minimal_mesh_t
   static constexpr size_t dimension = minimal_config_t::num_dimensions;
   
   using vertex_t = minimal_types_t::vertex_t;
+  using edge_t = minimal_types_t::edge_t;
+#if FLECSI_MESH_DIMENSION == 3
```
Comment (junghans) [inline flecsi-sp/minimal/mesh.h]: Where can I define `FLECSI_MESH_DIMENSION` to be `3`? (It is hard-coded to `2` in `specializations/minimal/config.h`.

File: config/project.cmake (hunk 2 above)
Comment (ryosukep) [inline config/project.cmake]: All "#include" statement of flecsi-sp uses "flecsi-sp" as a parent-dir, not "specializations". It seems like that was due to old-version of cinch created symbolic link to the header inside build directory but created the directory such as "build/include/flecsi-sp" that is linked to "flecsi-sp/specializations". This symlink directory no longer exists during build with newer cinch. 

The easiest way to mitigate this non-backward compatibility of the cinch is to simply change the directory structure of flecsi-sp to similar to other packages that use newer cinch, such as flecsi and flecsale. These packages appears to have a directory structure like "flecsi/flecsi" and "flecsale/flecsale". 

If this is a problem, then please guide me through how I should fix them.

Comment (charest) [inline config/project.cmake]: I think I would keep `cinch_add_library_target(flecsi-sp specializations)`, and just replace `#include "flecsi-sp/...."` with `#include "specialiations/...."`

Comment (charest) [inline config/project.cmake]: I'm not sure what effect this will have on projects that use `flecsi-sp` as an installed library though...

**Pull request closed without merging**